mssw/My-List-Util/lib/Test/My/List/Util.pm

```diff
@@ -0,0 +1,227 @@
+package Test::My::List::Util;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Exporter qw(import);
+use vars qw(@EXPORT $VERSION);
+
+use Test::Builder;
+
+my $Test = new Test::Builder();
+
+$VERSION = '3.14';
+@EXPORT  = qw(sum_ok);
+
+=head1 NAME
+
+Test::My::List::Util - Test utility module for My::List::Util
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 SYNOPSIS
+
+Helps with testing My::List::Util
+
+=head1 EXPORT
+
+A list of functions that can be exported.  You can delete this section
+if you dont export anything, such as for a purely object-oriented module.
+
+=head1 SUBROUTINES/METHODS
+
+=head2 sum_ok
+
+checks that a list can sum correctly.
+
+    sum_ok( $actual, $expected );
+    sum_ok( $actual, $expected, "The sums don't match!" );
+
+=cut
+
+sub sum_ok {
+  my ( $actual, $expected, $message ) = @_;
+
+  if ( !defined $actual && !defined $expected ) {
+    $Test->ok( 1, $message );
+    return 1;
+  }
+  elsif ( !defined $actual ) {
+    $Test->diag( "Actual value was not defined, expected [$expected]!\n" );
+    $Test->ok( 0, $message );
+    return 0;
+  }
+  elsif ( !defined $expected ) {
+    $Test->diag( "Actual value was [$actual], expected undefined!\n" );
+    $Test->ok( 0, $message );
+    return 0;
+  }
+  else {
+    # We have two defined values now.
+    # Lets see if they're both scalars
+    my $actual_ref   = ref $actual;
+    my $expected_ref = ref $expected;
+
+    if ( $actual_ref && $expected_ref ) {
+
+      # we have two references for some reason....
+      if ( $actual_ref eq $expected_ref ) {
+
+        # They're both the same type of reference, so lets just say they're the same.
+        $Test->diag( "Returning ok for two matching reference types - [$actual_ref]!\n" );
+        $Test->ok( 1, $message );
+        return 1;
```

Comment:
But this test function wants numeric sums, not refs. Also you could pass in two hashrefs with differing contents and pass this test.